Projects/Profile/Sources/Navigation/ProfileNavigation.swift

```diff
@@ -0,0 +1,88 @@
+import SwiftUI
+import hCore
+import hCoreUI
+
+public class ProfileNavigationViewModel: ObservableObject {
+    @Published var isDeleteAccountPresented: MemberDetails?
+    @Published var isDeleteAccountAlreadyRequestedPresented = false
+    @Published var isLanguagePickerPresnted = false
+}
+
+public enum ProfileNavigationDismissAction {
+    case openChat
+}
+
+public struct ProfileNavigation<Content: View>: View {
+    @ViewBuilder var redirect: (_ type: ProfileRedirectType) -> Content
+    @StateObject var router = Router()
+    @StateObject var profileNavigationViewModel = ProfileNavigationViewModel()
+
+    public init(@ViewBuilder redirect: @escaping (_ type: ProfileRedirectType) -> Content) {
+        self.redirect = redirect
+    }
+
+    public var body: some View {
+        RouterHost(router: router) {
+            ProfileView()
+                .routerDestination(for: ProfileRedirectType.self) { redirectType in
+                    switch redirectType {
+                    case .travelCertificate:
+                        redirect(.travelCertificate)
+                    case .myInfo:
+                        MyInfoView()
+                    case .appInfo:
+                        AppInfoView()
+                    case .settings:
+                        SettingsScreen()
+                    case .euroBonus:
+                        EuroBonusNavigation()
+                    case .deleteAccount:
+                        EmptyView()
+                    case .pickLanguage:
+                        EmptyView()
+                    }
+                }
+        }
+        .environmentObject(router)
+        .environmentObject(profileNavigationViewModel)
+        .detent(
+            item: $profileNavigationViewModel.isDeleteAccountPresented,
+            style: .height,
+            options: .constant(.withoutGrabber)
+        ) { memberDetails in
+            redirect(
+                .deleteAccount(
+                    memberDetails: memberDetails
+                        //                onDismiss: {
```

Comment:
@sladan-hedvig this is where I tried to add dismiss closure to deleteAccount but it doesn't work. We need this to set profileNavigationViewModel.isDeleteAccountPresented = nil. Any idea on ay other approach?

case deleteAccount(memberDetails: MemberDetails, onDismiss: () -> Void)

Reply:
I pushed commit that should resolve this issue.
Basically having control on the profileNavigationViewModel from the both main and profile navigation.
Please give it a look. Something simular we have with the home vm